assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -12659,7 +12717,16 @@ public Time getCloseTime()
 		 */
 		public String getAnonymousSubmissionId() {
```

Comment:
This block seems very similar to the code on https://github.com/ox-it/sakai/pull/67/files#diff-36d7daf9ff6912ed634cfd484790f5d8R11610 could we extract this out to a method as the logic is more complicated now.